File: RefugeUA.WebApp/refugeua.webapp.client/src/app/core/models/psychologist-information.ts
import { Entity } from "./entity"
import { ContactInformation } from "./contact-information"
import { User } from "./user"

export interface PsychologistInformation extends Entity {
    title: string,
    description: string,
    createdAt: Date,
    authorId?: number,
    author?: User,
    contactId: number,
    contact?: ContactInformation
}

File: RefugeUA.Entities/PsychologistInformation.cs
using RefugeUA.Entities.Abstracts;
using RefugeUA.Entities.Interfaces;

namespace RefugeUA.Entities
{
    public class PsychologistInformation : Entity
    {
        public PsychologistInformation()
        {
            this.CreatedAt = DateTime.Now;
        }

        public string Title { get; set; } = default!;

        public string Description { get; set; } = default!;

        public DateTime CreatedAt { get; set; }

        public long? AuthorId { get; set; }

        public IUser Author { get; set; } = default!;

        public long ContactId { get; set; }

        public ContactInformation Contact { get; set; } = default!;
    }
}
